tests/client/amo/containers/TestSearch.js

```diff
@@ -0,0 +1,165 @@
+import * as searchActions from 'core/actions/search';
+import {
+  isLoaded,
+  loadSearchResultsIfNeeded,
+  mapStateToProps,
+  parsePage,
+} from 'amo/containers/Search';
+import * as api from 'core/api';
+
+describe('Search.mapStateToProps()', () => {
+  const state = {
+    addons: { ab: { slug: 'ab', name: 'ad-block' },
+             cd: { slug: 'cd', name: 'cd-block' } },
+    search: { query: 'ad-block', loading: false, results: [{ slug: 'ab', name: 'ad-block' }] },
+  };
+
+  it('passes the search state if the URL and state query matches', () => {
+    const props = mapStateToProps(state, { location: { query: { q: 'ad-block' } } });
+    assert.strictEqual(props, state.search);
+  });
+
+  it('does not pass search state if the URL and state query do not match', () => {
+    const props = mapStateToProps(state, { location: { query: { q: 'more-ads' } } });
+    assert.deepEqual(props, {});
+  });
+});
+
+describe('Search.isLoaded()', () => {
+  const state = {
+    page: 2,
+    query: 'ad-block',
+    loading: false,
+    results: [{ slug: 'ab', name: 'ad-block' }],
+  };
+
+  it('is loaded when not loading and page and query page', () => {
+    assert(isLoaded({ state, page: 2, query: 'ad-block' }));
+  });
+
+  it('is not loaded when loading', () => {
+    assert(!isLoaded({
+      state: Object.assign({}, state, { loading: true }),
+      page: 2,
+      query: 'ad-block',
+    }));
+  });
+
+  it('is not loaded when the query does not match', () => {
+    assert(!isLoaded({ state, page: 2, query: 'youtube' }));
+  });
+
+  it('is not loaded when the page does not match', () => {
+    assert(!isLoaded({ state, page: 3, query: 'ad-block' }));
+  });
+});
+
+describe('CurrentSearchPage.parsePage()', () => {
+  it('returns a number', () => {
+    assert.strictEqual(parsePage(10), 10);
+  });
+
+  it('parses a number from a string', () => {
+    assert.strictEqual(parsePage('8'), 8);
+  });
+
+  it('treats negatives as 1', () => {
+    assert.strictEqual(parsePage('-10'), 1);
+  });
+
+  it('treats words as 1', () => {
+    assert.strictEqual(parsePage('hmmm'), 1);
+  });
+
+  it('treats "0" as 1', () => {
+    assert.strictEqual(parsePage('0'), 1);
+  });
+
+  it('treats 0 as 1', () => {
+    assert.strictEqual(parsePage(0), 1);
+  });
+
+  it('treats empty strings as 1', () => {
+    assert.strictEqual(parsePage(''), 1);
+  });
+});
+
+describe('CurrentSearchPage.loadSearchResultsIfNeeded()', () => {
+  it('does not dispatch on undefined query', () => {
+    const dispatchSpy = sinon.spy();
+    const state = { loading: false };
+    const store = { dispatch: dispatchSpy, getState: () => ({ search: state }) };
```

Comment:
Could put each prop on a new line, make it a bit easier to read + saves on scanning horizontally.